ngi_reports/stockholm/project_summary.py

```diff
@@ -76,94 +80,120 @@ def __init__(self, config, LOG, working_dir, **kwargs):
+                    keys = sorted(lib_valids.keys(), key=lambda k: datetime.strptime(lib_valids[k]['start_date'], "%Y-%m-%d"), reverse=True)
+                    self.samples_info[sample_id]['preps'][prep_id]['avg_size'] = lib_valids[keys[0]]['average_size_bp']
+                except KeyError:
+                    self.samples_info[sample_id]['preps'][prep_id]['avg_size'] = ''
+            
+        ## Collect reuired information for all flowcell run for the project
+        for fc in self.flowcell_info.values():
+            fc_nm = fc['name']
+            fc_obj = fcon.get_entry(fc['run_name'])
             fc_runp = fc_obj.get('RunParameters',{})
+            fc_illumina = fc_obj.get('illumina',{})
+            fc_run_summary = fc_illumina.get('run_summary',{})
+            self.flowcell_info[fc_nm]['lanes'] = {}
+            
+            ## Get sequecing method for the flowcell
+            seq_template = "{}) Clustering was done by '{}' and samples were sequenced on {} ({}) with a {} setup using '{}' "\
+                           "chemistry. The Bcl to Fastq conversion was performed using {} from the CASAVA software suite. The "\
+                           "quality scale used is Sanger / phred33 / Illumina 1.8+."
+            run_setup = fc_obj.get("run_setup")
+            fc_chem = fc_runp.get('ReagentKitVersion') if fc_runp.get('ReagentKitVersion') else fc_runp.get('Sbs')
             seq_plat = ["HiSeq2500","MiSeq"]["MCSVersion" in fc_runp.keys()]
+            clus_meth = ["cBot","onboard clustering"][seq_plat == "MiSeq" or fc_runp.get("ClusteringChoice","") == "OnBoardClustering"]
             casava = fc_obj.get('DemultiplexConfig',{}).values()[0].get('Software',{}).get('Version')
             if seq_plat == "MiSeq":
                 seq_software = "MSC {}/RTA {}".format(fc_runp.get("MCSVersion"),fc_runp.get("RTAVersion"))
             else:
                 seq_software = "{} {}/RTA {}".format(fc_runp.get("ApplicationName"),fc_runp.get("ApplicationVersion"),fc_runp.get("RTAVersion"))
-            tmp_method = seq_template.format("SECTION", seq_plat, seq_software, run_setup,casava)
+            tmp_method = seq_template.format("SECTION", clus_meth, seq_plat, seq_software, run_setup, fc_chem, casava)
             ## to make sure the sequencing methods are uniq
-            if tmp_method not in seq_method_info:
-                seq_method_info.append(tmp_method)
-
+            if tmp_method not in seq_methods.keys():
+                seq_methods[tmp_method] = alphabets[len(seq_methods.keys())]
+            self.flowcell_info[fc_nm]['seq_meth'] = seq_methods[tmp_method]
+            
+            ## Collect quality info for samples and collect lanes of interest
+            for stat in fc_illumina.get('Demultiplex_Stats',{}).get('Barcode_lane_statistics',[]):
+                try:
+                    sample, lane = (stat['Sample ID'], stat['Lane'])
+                    if stat['Project'].replace('__','.') != self.project_name:
```

Comment:
Yup, it is '__' (double underscore), dont ask me why :stuck_out_tongue_winking_eye: